src/dataflow-types/src/proto/logging.rs

```diff
@@ -0,0 +1,240 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+include!(concat!(env!("OUT_DIR"), "/logging.rs"));
+
+use std::collections::HashMap;
+
+use crate::logging::{DifferentialLog, LogVariant, LoggingConfig, MaterializedLog, TimelyLog};
+use mz_expr::GlobalId;
+use mz_repr::proto::{ProtoRepr, TryFromProtoError, TryIntoIfSome};
+
+impl From<&LoggingConfig> for ProtoLoggingConfig {
+    fn from(x: &LoggingConfig) -> Self {
+        ProtoLoggingConfig {
+            granularity_ns: Some(x.granularity_ns.into_proto()),
+            active_logs: x.active_logs.iter().map(Into::into).collect(),
+            log_logging: x.log_logging,
+        }
+    }
+}
+
+impl TryFrom<ProtoLoggingConfig> for LoggingConfig {
+    type Error = TryFromProtoError;
+
+    fn try_from(x: ProtoLoggingConfig) -> Result<Self, Self::Error> {
+        Ok(LoggingConfig {
+            granularity_ns: x.granularity_ns.map(u128::from_proto).ok_or_else(|| {
+                TryFromProtoError::missing_field("ProtoLoggingConfig::granularity_ns")
+            })??,
+            active_logs: x
+                .active_logs
+                .into_iter()
+                .map(TryInto::try_into)
+                .collect::<Result<HashMap<_, _>, _>>()?,
+            log_logging: x.log_logging,
+        })
+    }
+}
+
+impl From<(&LogVariant, &GlobalId)> for ProtoActiveLog {
+    fn from(x: (&LogVariant, &GlobalId)) -> Self {
+        ProtoActiveLog {
+            key: Some(x.0.into()),
+            value: Some(x.1.into()),
+        }
+    }
+}
+
+impl TryFrom<ProtoActiveLog> for (LogVariant, GlobalId) {
+    type Error = TryFromProtoError;
+
+    fn try_from(x: ProtoActiveLog) -> Result<Self, Self::Error> {
+        Ok((
+            x.key.try_into_if_some("ProtoActiveLog::key")?,
+            x.value.try_into_if_some("ProtoActiveLog::value")?,
+        ))
+    }
+}
+
+impl From<&LogVariant> for ProtoLogVariant {
+    fn from(x: &LogVariant) -> Self {
+        ProtoLogVariant {
+            kind: Some(match x {
+                LogVariant::Timely(timely) => proto_log_variant::Kind::Timely(timely.into()),
+                LogVariant::Differential(differential) => {
+                    proto_log_variant::Kind::Differential(differential.into())
+                }
+                LogVariant::Materialized(materialize) => {
+                    proto_log_variant::Kind::Materialized(materialize.into())
+                }
+            }),
+        }
+    }
+}
+
+impl TryFrom<ProtoLogVariant> for LogVariant {
+    type Error = TryFromProtoError;
+
+    fn try_from(x: ProtoLogVariant) -> Result<Self, Self::Error> {
```

Comment:
nit: the convention in most of the other implementations is to import

```rust
use ${camel_cased_t}::Kind::*;
```

at the top of `try_from` and `from` for `enum` types `$T`.

If `Proto$T` itself defines nested types to handle some of its variants, the `fn from(..)` can start with:

```rust
use ${camel_cased_t}::*;
use Kind::*;
```

That makes the case handling a bit shorter, and due to the simplicity of most cases so far I have not seen a case where it causes naming collisions.

Reply:
I referenced `src/expr/src/proto/id.rs`, which happened to the only file that did not conform to this convention. Any particular reason why it doesn't conform?

Reply:
I think it was the first PR from @lluki and I failed to catch that.